core/templates/dev/head/pages/collection_editor/editor_tab/CollectionLinearizerService.js

```diff
@@ -205,12 +208,15 @@ oppia.factory('CollectionLinearizerService', [
           return false;
         }
 
+        var linearNodeList = _getCollectionNodesInPlayableOrder(collection);
+        var nodeIndex = findNodeIndex(linearNodeList, explorationId);
+        var collectionNode = linearNodeList[nodeIndex];
+        var acquiredSkill =
+          collectionNode.getAcquiredSkillList().getSkills()[0];
```

Comment:
I think the only guarantee that we're using is the linearity of the structure (there's only 1 element in the array), and that each acquired skill is only given by one node (so we should delete the acquired skill). The only way these assumptions would seem to change is if the concept of a "linearlizer" changes.